File: Day 006/2491. Divide Players Into Teams of Equal Skill.java
class Solution {
    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);
        int right = skill.length - 1;
        int left = 0;
        int a = skill[0] + skill[skill.length - 1];
        long re = 0;
        while (left < right) {
            if ((skill[left] + skill[right]) == a) {
                re += (skill[left]) * skill[right];
                left++;
                right--;
            } else {
                return -1;
            }
        }
        return re;
    }
}

File: Day 006/2491. Divide Players Into Teams of Equal Skill.py
class Solution(object):
    def dividePlayers(self, skill):
        skill.sort()
        right = len(skill) - 1;
        left = 0;
        a = skill[0] + skill[len(skill) - 1];
        re =0
        while left < right:
            if (skill[left] + skill[right]) == a:
                re += (skill[left]) * skill[right];
                left+=1;
                right-=1;
            else:
                return -1
            
        return re
        """
        :type skill: List[int]
        :rtype: int
        """
